environments/templates/auxtel-application.yaml

```diff
@@ -0,0 +1,53 @@
+{{- if (index .Values "applications" "auxtel") -}}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: auxtel
+spec:
+  finalizers:
+    - kubernetes
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: auxtel
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  destination:
+    namespace: auxtel
+    server: https://kubernetes.default.svc
+  project: default
+  source:
+    path: applications/auxtel
+    repoURL: {{ .Values.repoURL }}
+    targetRevision: {{ .Values.targetRevision }}
+    helm:
+      parameters:
+        - name: "global.host"
+          value: {{ .Values.fqdn | quote }}
+        - name: "global.baseUrl"
+          value: "https://{{ .Values.fqdn }}"
+        - name: "global.vaultSecretsPath"
+          value: {{ .Values.vaultPathPrefix | quote }}
+        - name: "global.controlSystem.appNamespace"
+          value: "auxtel"
+        - name: "global.controlSystem.imageTag"
+          value: {{ .Values.controlSystem.imageTag | quote }}
+        - name: "global.controlSystem.siteTag"
+          value: {{ .Values.controlSystem.siteTag | quote }}
+        - name: "global.controlSystem.topicName"
+          value: {{ .Values.controlSystem.topicName | quote }}
+        - name: "global.controlSystem.kafkaBrokerAddress"
+          value: {{ .Values.controlSystem.kafkaBrokerAddress | quote }}
+        - name: "global.controlSystem.kafkaTopicReplicationFactor"
+          value: {{ .Values.controlSystem.kafkaTopicReplicationFactor | quote }}
+        - name: "global.controlSystem.schemaRegistryUrl"
+          value: {{ .Values.controlSystem.schemaRegistryUrl | quote }}
+        - name: "global.controlSystem.s3EndpointUrl"
+          value: {{ .Values.controlSystem.s3EndpointUrl | quote }}
```

Comment:
These injections will also need to be added to `phalanx.services.ApplicationService._build_injected_values` so that `phalanx application lint` and `phalanx application template` work correctly.

Reply:
OK. Should I just inject them into the dictionary or trap them going in? Trapping means specifying a long list of apps to check which seems unsatisfactory.

Reply:
Eh, the more I dig at this, the more confused I get. I think I'm going to need some help getting this sorted out.

Reply:
Something like this:
```python
        if environment.control_system:
            extra = {
                "appNamespace": environment.control_system.app_namespace,
                "imageTag": environment.control_system.image_tag,
                "siteTag": environment.control_system.site_tag,
                "topicName": environment.control_system.topic_name,
                "kafkaBrokerAddress": (
                    environment.control_system.kafka_broker_address
                ),
                "kafkaTopicReplicationFactor": (
                    environment.control_system.kafka_topic_replication_factor
                ),
                "schemaRegistryUrl": (
                    environment.control_system.schema_registry_url
                ),
                "s3EndpointUrl": environment.control_system.s3_endpoint_url,
            }
            values.extend(
                {
                    f"global.controlSystem.{k}": v
                    for k, v in extra.items()
                    if v is not None
                }
            )
```
It's okay to inject them for every application.

Reply:
It looks like `control_system` moved back into `EnvironmentConfig` somewhere along the line. It needs to be in `EnvironmentBaseConfig`.

Reply:
Arg. That was me not paying attention. Sigh.

Reply:
OK. I think I now have this working. The only thing that doesn't "work" is `appNamespace` in something like `phalanx application template`. That particular parameter is filled in the application template over in `environments/templates` which that command ignores. As the docs say, this does't quite reproduce Argo CD behavior, so I think this is good enough.